### ISSUE TITLE
Any way to install with CUDA 9.0?

### ISSUE THREAD

**Issue by seankala** [rusty1s/pytorch_scatter#125]
Hello. I've been trying to install `torch_scatter` but have been having some problems. The CUDA version that I use is 9.0 but I noticed that the lowest you provide is 9.2. I could always download the 9.2 version and it downloads fine, but when I run `import torch_scatter` I get an error.

**Comment by rusty1s** [comment]
What exactly is the error?

**Comment by seankala** [comment]
The error was something like `ImportError: libcudart.so.9.2`. Sorry for the vagueness, my computer shut down and I wasn't able to write down the error message properly.

I don't know if this helps, but right now I have PyTorch 1.4.0 installed with CUDA 9.0. One of my labmates recommended that I try to install PyTorch 1.1.0 and find a `torch_scatter` that's "compatible to that version."

**Comment by rusty1s** [comment]
Well, you can also try installing torch-scatter from source. That should work too.

**Comment by seankala** [comment]
I've tried that as well, and I get a very length error message that's all in red. I don't recall the error message. I'll try it again and come back with the results and perhaps we'll be able to move on or close this issue then. :)

**Comment by seankala** [comment]
So I installed PyTorch 1.1.0 and tried running `pip install torch-scatter` after setting the path environmental variables correctly and got the following error script:

```
Collecting torch-scatter
  Using cached torch_scatter-2.0.4.tar.gz (20 kB)
Building wheels for collected packages: torch-scatter
  Building wheel for torch-scatter (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /home/seokwon/anaconda3/envs/seanconda/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_95f9aoo/torch-scatter/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_95f9aoo/torch-scatter/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-p373da14
       cwd: /tmp/pip-install-_95f9aoo/torch-scatter/
  Complete output (101 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/torch_scatter
  copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter
  copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.7/torch_scatter
  copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.7/torch_scatter
  copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.7/torch_scatter
  copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.7/torch_scatter
  copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.7/torch_scatter
  creating build/lib.linux-x86_64-3.7/torch_scatter/composite
  copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
  copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
  copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
  copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
  running build_ext
  building 'torch_scatter._segment_csr' extension
  creating build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/tmp
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc/cpu
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc/cuda
  gcc -pthread -B /home/seokwon/anaconda3/envs/seanconda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/tmp/pip-install-_95f9aoo/torch-scatter/csrc -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/TH -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/seokwon/anaconda3/envs/seanconda/include/python3.7m -c /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp -o build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_segment_csr -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_segment_coo -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_version -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_scatter -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++11
  cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
  <command-line>:0:0: warning: "TORCH_EXTENSION_NAME" redefined
  <command-line>:0:0: note: this is the location of the previous definition
  <command-line>:0:0: warning: "TORCH_EXTENSION_NAME" redefined
  <command-line>:0:0: note: this is the location of the previous definition
  <command-line>:0:0: warning: "TORCH_EXTENSION_NAME" redefined
  <command-line>:0:0: note: this is the location of the previous definition
  In file included from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:5:0:
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:6:49: error: ‘List’ in namespace ‘c10’ does not name a template type
   inline std::vector<int64_t> list2vec(const c10::List<int64_t> list) {
                                                   ^
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:6:53: error: expected ‘,’ or ‘...’ before ‘<’ token
   inline std::vector<int64_t> list2vec(const c10::List<int64_t> list) {
                                                       ^
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h: In function ‘std::vector<long int> list2vec(int)’:
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:8:18: error: ‘list’ was not declared in this scope
     result.reserve(list.size());
                    ^
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:8:18: note: suggested alternatives:
  In file included from /usr/include/c++/5/list:63:0,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/autograd/variable.h:12,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/autograd/generated/variable_factories.h:8,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/types.h:7,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/script.h:3,
                   from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:2:
  /usr/include/c++/5/bits/stl_list.h:507:11: note:   ‘std::list’
       class list : protected _List_base<_Tp, _Alloc>
             ^
  In file included from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/ivalue.h:7:0,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/stack.h:3,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/jit/tracer.h:5,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/autograd/generated/variable_factories.h:9,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/types.h:7,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/script.h:3,
                   from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:2:
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:124:11: note:   ‘c10::aten::list’
     _(aten, list)                    \
             ^
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:322:35: note: in definition of macro ‘DEFINE_SYMBOL’
     namespace ns { constexpr Symbol s(static_cast<unique_t>(_keys::ns##_##s)); }
                                     ^
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:323:1: note: in expansion of macro ‘FORALL_NS_SYMBOLS’
   FORALL_NS_SYMBOLS(DEFINE_SYMBOL)
   ^
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:124:11: note:   ‘c10::aten::list’
     _(aten, list)                    \
             ^
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:322:35: note: in definition of macro ‘DEFINE_SYMBOL’
     namespace ns { constexpr Symbol s(static_cast<unique_t>(_keys::ns##_##s)); }
                                     ^
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:323:1: note: in expansion of macro ‘FORALL_NS_SYMBOLS’
   FORALL_NS_SYMBOLS(DEFINE_SYMBOL)
   ^
  In file included from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/cast.h:13:0,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/attr.h:13,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/pybind11.h:44,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/utils/pybind.h:6,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/python.h:12,
                   from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/extension.h:6,
                   from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/cpu/segment_csr_cpu.h:3,
                   from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:4:
  /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/pytypes.h:1236:7: note:   ‘pybind11::list’
   class list : public object {
         ^
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp: At global scope:
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:43:24: error: ‘torch::autograd::AutogradContext’ has not been declared
   using torch::autograd::AutogradContext;
                          ^
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:47:55: error: expected template-name before ‘<’ token
   class SegmentSumCSR : public torch::autograd::Function<SegmentSumCSR> {
                                                         ^
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:47:55: error: expected ‘{’ before ‘<’ token
  /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:47:55: error: expected unqualified-id before ‘<’ token
  error: command 'gcc' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for torch-scatter
  Running setup.py clean for torch-scatter
Failed to build torch-scatter
Installing collected packages: torch-scatter
    Running setup.py install for torch-scatter ... error
    ERROR: Command errored out with exit status 1:
     command: /home/seokwon/anaconda3/envs/seanconda/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_95f9aoo/torch-scatter/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_95f9aoo/torch-scatter/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-qpysu5wv/install-record.txt --single-version-externally-managed --compile --install-headers /home/seokwon/anaconda3/envs/seanconda/include/python3.7m/torch-scatter
         cwd: /tmp/pip-install-_95f9aoo/torch-scatter/
    Complete output (101 lines):
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.7
    creating build/lib.linux-x86_64-3.7/torch_scatter
    copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter
    copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.7/torch_scatter
    copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.7/torch_scatter
    copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.7/torch_scatter
    copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.7/torch_scatter
    copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.7/torch_scatter
    creating build/lib.linux-x86_64-3.7/torch_scatter/composite
    copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
    copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
    copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
    copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.7/torch_scatter/composite
    running build_ext
    building 'torch_scatter._segment_csr' extension
    creating build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/tmp
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc/cpu
    creating build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc/cuda
    gcc -pthread -B /home/seokwon/anaconda3/envs/seanconda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/tmp/pip-install-_95f9aoo/torch-scatter/csrc -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/TH -I/home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/THC -I/usr/local/cuda/include -I/home/seokwon/anaconda3/envs/seanconda/include/python3.7m -c /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp -o build/temp.linux-x86_64-3.7/tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_segment_csr -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_segment_coo -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_version -D_GLIBCXX_USE_CXX11_ABI=0 -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_scatter -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++11
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++
    <command-line>:0:0: warning: "TORCH_EXTENSION_NAME" redefined
    <command-line>:0:0: note: this is the location of the previous definition
    <command-line>:0:0: warning: "TORCH_EXTENSION_NAME" redefined
    <command-line>:0:0: note: this is the location of the previous definition
    <command-line>:0:0: warning: "TORCH_EXTENSION_NAME" redefined
    <command-line>:0:0: note: this is the location of the previous definition
    In file included from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:5:0:
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:6:49: error: ‘List’ in namespace ‘c10’ does not name a template type
     inline std::vector<int64_t> list2vec(const c10::List<int64_t> list) {
                                                     ^
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:6:53: error: expected ‘,’ or ‘...’ before ‘<’ token
     inline std::vector<int64_t> list2vec(const c10::List<int64_t> list) {
                                                         ^
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h: In function ‘std::vector<long int> list2vec(int)’:
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:8:18: error: ‘list’ was not declared in this scope
       result.reserve(list.size());
                      ^
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/utils.h:8:18: note: suggested alternatives:
    In file included from /usr/include/c++/5/list:63:0,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/autograd/variable.h:12,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/autograd/generated/variable_factories.h:8,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/types.h:7,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/script.h:3,
                     from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:2:
    /usr/include/c++/5/bits/stl_list.h:507:11: note:   ‘std::list’
         class list : protected _List_base<_Tp, _Alloc>
               ^
    In file included from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/ivalue.h:7:0,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/stack.h:3,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/jit/tracer.h:5,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/autograd/generated/variable_factories.h:9,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/types.h:7,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/script.h:3,
                     from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:2:
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:124:11: note:   ‘c10::aten::list’
       _(aten, list)                    \
               ^
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:322:35: note: in definition of macro ‘DEFINE_SYMBOL’
       namespace ns { constexpr Symbol s(static_cast<unique_t>(_keys::ns##_##s)); }
                                       ^
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:323:1: note: in expansion of macro ‘FORALL_NS_SYMBOLS’
     FORALL_NS_SYMBOLS(DEFINE_SYMBOL)
     ^
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:124:11: note:   ‘c10::aten::list’
       _(aten, list)                    \
               ^
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:322:35: note: in definition of macro ‘DEFINE_SYMBOL’
       namespace ns { constexpr Symbol s(static_cast<unique_t>(_keys::ns##_##s)); }
                                       ^
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/ATen/core/interned_strings.h:323:1: note: in expansion of macro ‘FORALL_NS_SYMBOLS’
     FORALL_NS_SYMBOLS(DEFINE_SYMBOL)
     ^
    In file included from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/cast.h:13:0,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/attr.h:13,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/pybind11.h:44,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/utils/pybind.h:6,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/csrc/api/include/torch/python.h:12,
                     from /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/torch/extension.h:6,
                     from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/cpu/segment_csr_cpu.h:3,
                     from /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:4:
    /home/seokwon/anaconda3/envs/seanconda/lib/python3.7/site-packages/torch/include/pybind11/pytypes.h:1236:7: note:   ‘pybind11::list’
     class list : public object {
           ^
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp: At global scope:
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:43:24: error: ‘torch::autograd::AutogradContext’ has not been declared
     using torch::autograd::AutogradContext;
                            ^
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:47:55: error: expected template-name before ‘<’ token
     class SegmentSumCSR : public torch::autograd::Function<SegmentSumCSR> {
                                                           ^
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:47:55: error: expected ‘{’ before ‘<’ token
    /tmp/pip-install-_95f9aoo/torch-scatter/csrc/segment_csr.cpp:47:55: error: expected unqualified-id before ‘<’ token
    error: command 'gcc' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/seokwon/anaconda3/envs/seanconda/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-_95f9aoo/torch-scatter/setup.py'"'"'; __file__='"'"'/tmp/pip-install-_95f9aoo/torch-scatter/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-qpysu5wv/install-record.txt --single-version-externally-managed --compile --install-headers /home/seokwon/anaconda3/envs/seanconda/include/python3.7m/torch-scatter Check the logs for full command output.
```

Any ideas or suggestions on how I can go about this?

**Comment by rusty1s** [comment]
Yes, this is the usual error you get when you are working on PyTorch < 1.4.

**Comment by seankala** [comment]
Is there a compatible version of `torch-scatter` that I could use to work with PyTorch 1.1.0? I'm being a bit stubborn about the version because apparently the lab server administrator refuses to upgrade the CUDA versions and they're all 9.0.

**Comment by seankala** [comment]
I'm checking out the [history](https://pypi.org/project/torch-scatter/#history) for the versions but am not sure which one I should download.

**Comment by rusty1s** [comment]
From 1.2 to 1.4, see [here](https://github.com/rusty1s/pytorch_scatter/releases).

**Comment by rusty1s** [comment]
Btw, you do not need CUDA development kit to install `torch-scatter` from wheel.

**Comment by seankala** [comment]
Thanks! I managed to install `torch-scatter` after specifying the versions you mentioned. This may be a bit of a different issue, but I actually need the `scatter` module from `torch-scatter`, and I noticed that I can't import it (`ImportError: cannot import name 'scatter' from 'torch_scatter' (/home/user/anaconda3/envs/userconda/lib/python3.7/site-packages/torch_scatter/__init__.py)`). Is this because the version of `torch-scatter` that I installed is too low?
